digitalocean/resource_digitalocean_image.go

```diff
@@ -179,15 +179,28 @@ func resourceDigitalOceanCustomImageUpdate(ctx context.Context, d *schema.Resour
 		return diag.Errorf("Error converting id %s to string: %s", imageID, err)
 	}
 
-	if d.HasChange("name") {
-		name := d.Get("name").(string)
-		_, _, err := client.Images.Update(ctx, id, &godo.ImageUpdateRequest{
-			Name: name,
-		})
+	if d.HasChanges("name", "description", "distribution") {
+		imageName := d.Get("name").(string)
+		imageUpdateRequest := &godo.ImageUpdateRequest{}
+
+		if d.HasChange("name") {
+			imageUpdateRequest.Name = imageName
+		}
+		if d.HasChange("distribution") {
+			imageDistribution := d.Get("distribution").(string)
+			imageUpdateRequest.Distribution = imageDistribution
+		}
+
+		if d.HasChange("description") {
+			imageDescription := d.Get("description").(string)
+			imageUpdateRequest.Description = imageDescription
+		}
```

Comment:
```suggestion
		imageUpdateRequest := &godo.ImageUpdateRequest{
			Name:         imageName,
			Distribution: d.Get("distribution").(string),
			Description:  d.Get("description").(string),
		}

```

We can simplify this a bit. PUT is idempotent, so no reason not to build the full request.

Reply:
I didn't noticed it was idempotent, was trying to change only fields that changed. I'll make this change